src/mlpack/tests/ksinit_test.cpp

```diff
@@ -0,0 +1,315 @@
+/*
+  @file ksinit_test.cpp
+  @author Praveen Ch
+  
+
+  Tests the working of Kathirvalavakumar Subavathi Initialization for a 
+  Feed forward neural network.
+*/
+
+
+#include <mlpack/core.hpp>
+
+#include <mlpack/methods/ann/init_rules/random_init.hpp>
+#include <mlpack/methods/ann/init_rules/kathirvalavakumar_subavathi_init.hpp>
+
+#include <mlpack/methods/ann/ffn.hpp>
+#include <mlpack/core/optimizers/rmsprop/rmsprop.hpp>
+#include <mlpack/methods/ann/layer/layer.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "test_tools.hpp"
+
+using namespace mlpack;
+using namespace mlpack::ann;
+using namespace mlpack::optimization;
+
+BOOST_AUTO_TEST_SUITE(KSInitialization);
+
+/**
+ * Train and evaluate a vanilla network with the specified initialisation 
+   procedure.
+ */
+template<typename MatType = arma::mat>
+void BuildVanillaNetwork(MatType& trainData,
+                         MatType& trainLabels,
+                         MatType& testData,
+                         MatType& testLabels,
+                         const size_t hiddenLayerSize,
+                         const size_t maxEpochs,
+                         const size_t outputSize,
+                         double& trainError,
+                         double& testError)
+{
+  /*
+  @param trainError mean squared error of predictions on training data.
```

Comment:
Minor style issue, if you comment the parameters for the test, put the documentation over the function definition, that way  Doxygen can generate the documentation.